### PR TITLE
Handle Optionals safely

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Fixes the exception reported in https://github.com/PegaSysEng/teku/issues/1792

Handles node table lookups that don't find the entry in question (it may have been removed in the meantime safely) and ensure that the TaskManager executes recursive searches on it's task thread preserving the correct threading model.

### CODE REVIEW

File: src/main/java/org/ethereum/beacon/discovery/storage/NodeTableImpl.java [hunk 1]
```diff
@@ -163,35 +163,35 @@ private void loadNextBatch() {
...
+            upNodes -> {
+              for (Bytes currentNodeId : upNodes.getEntries()) {
+                if (logLimit == 0) {
+                  getNode(currentNodeId).ifPresent(res::add);
+                } else if (Functions.logDistance(currentNodeId, nodeId) >= logLimit) {
+                  limitReached = true;
+                  break;
+                } else {
+                  getNode(currentNodeId).ifPresent(res::add);
+                }
```
Comment (mbaxter) [inline src/main/java/org/ethereum/beacon/discovery/storage/NodeTableImpl.java:176]: (nit) Can this be simplified?:
```suggestion
                if (logLimit != 0 && Functions.logDistance(currentNodeId, nodeId) >= logLimit) {
                  limitReached = true;
                  break;
                } else {
                  getNode(currentNodeId).ifPresent(res::add);
                }
```

File: src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java [hunk 2]
```diff
@@ -229,9 +230,11 @@ private void liveCheckTask() {
...
 
   public CompletableFuture<Void> searchForNewPeers() {
-    return new RecursiveLookupTask(
-            nodeTable, this::findNodes, RECURSIVE_SEARCH_QUERY_LIMIT, Bytes32.random())
-        .execute();
+    final RecursiveLookupTask task =
+        new RecursiveLookupTask(
+            nodeTable, this::findNodes, RECURSIVE_SEARCH_QUERY_LIMIT, Bytes32.random());
+    // We wind up with a CompletableFuture<CompletableFuture> so unwrap one level.
+    return scheduler.execute(task::execute).thenCompose(Function.identity());
```
Comment (mbaxter) [inline src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java:237]: Why move this into the scheduler?  Looks like `searchForPeers` is already running inside of the scheduler most of the time: https://github.com/PegaSysEng/discovery/blob/0e8b14fe55d983dfe845dddad91b4ce940b20b0a/src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java#L155-L159

Maybe we need an internal method that gets scheduled at a fixed rate, and a separate public method that runs this in the scheduler?

Comment (ajsutton) [reply src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java:237]: True, good spot.

File: src/main/java/org/ethereum/beacon/discovery/storage/NodeTableImpl.java (hunk 1 above)
Comment (ajsutton) [reply src/main/java/org/ethereum/beacon/discovery/storage/NodeTableImpl.java]: Yep that works.

**Pull request merged**